src/core/ee/vif.cpp

```diff
@@ -278,6 +289,35 @@ void VectorInterface::init_UNPACK(uint32_t value)
     }
 }
 
+uint128_t VectorInterface::handle_UNPACKMasking(uint128_t quad)
+{
+    if (unpack.masked)
+    {
+        uint8_t tempmask;
+        
+        for (int i = 0; i < 4; i++) {
+            tempmask = (MASK >> ((i * 2) + std::min(unpack.blocks_written * 8, 24))) & 0x3;
+            if (tempmask == 1)
+            {
```

Comment:
Make this a switch statement (the compiler should theoretically optimize it into a jump table, and plus it's a bit cleaner)